server/resp/src/main/java/org/infinispan/server/resp/commands/ArgumentUtils.java

```diff
@@ -12,7 +10,29 @@ public final class ArgumentUtils {
    private ArgumentUtils() {
 
    }
+
    public static long toLong(byte[] argument) {
-      return Long.parseLong(new String(argument, StandardCharsets.US_ASCII));
+      if (argument == null || argument.length == 0)
+         throw new NumberFormatException("Empty argument");
+
+      boolean negative = false;
+      int i = 0;
+      if (argument[0] < '0') {
+         if (argument[0] != '-')
+            throw new NumberFormatException("Invalid character: " + argument[0]);
+
+         negative = true;
+         i = 1;
+      }
+
+      long result;
+      for (result = 0L; i < argument.length; i++) {
+         byte b = argument[i];
+         if (b < '0' || b > '9')
+            throw new NumberFormatException("Invalid character: " + b);
+
+         result = (result << 3) + (result << 1) + (b - 48);
+      }
```

Comment:
you can save 1 loop and some bitshift if needed. Maybe it could have some impact on small values

```suggestion
         
      byte b = argument[i++];
      if (b < '0' || b > '9')
         throw new NumberFormatException("Invalid character: " + b);
      result = (b - 48);
         
      for (; i < argument.length; i++) {
         b = argument[i];
         if (b < '0' || b > '9')
            throw new NumberFormatException("Invalid character: " + b);

         result = (result << 3) + (result << 1) + (b - 48);
      }
```